testrunner-maven-plugin/src/main/scala/edu/illinois/cs/testrunner/mavenplugin/TestPluginPlugin.scala

```diff
@@ -95,47 +73,10 @@ class TestPluginPlugin extends AbstractMojo {
 //
 //        println(configuredSurefire)
 
-        TestPluginPlugin.mojo = this
-        TestPluginPlugin.mavenProject = project
+        TestPluginUtil.logger = Logger.getLogger(className)
```

Comment:
Please do not change the logger that is used by the Maven plugin (e.g., the logger from AbstractMojo vs. java.util.logging.Logger)

In general, minimizing the changes to the Maven plugin unless they improve the Maven plugin or is needed for the Gradle plugin would help

Reply:
But the thing is Gradle cannot support the logger from AbstractMojo. In this way, the client code (e.g. iDFlakies) should be aware of whether the project is gradle project or maven project and chooses different loggers from TestPluginUtil. Right now, the gradle/maven differences are hidden/transparent to client code inside the ProjectWrapper and TestPluginUtil.

BTW, I reply to you about the above major comment in the email. Our goal is to make a major version change (e.g. from 1.2.0-2.0.0) for testrunner and iDFlakies so that gradle/maven transparency and easier future development and maintenance is our primary goal for this PR.

Reply:
Can the testrunner changes not check if the plugin is a Maven plugin and then use the logger from AbstractMojo? Changing the format of the log output is quite undesirable

The goal here is not for one iDFlakies version (e.g., a version with your iDFlakies PR changes) to work on older versions of testrunner, but the other way around, for one testrunner version (this version with your changes) to work with many versions of iDFlakies/iFixFlakies.

If this version of testrunner is used with an iDFlakies version that is only using MavenProject, testrunner should treat it as a Maven plugin as it did without these changes. As I mentioned in my last major comment, prioritizing less on future development/maintenance is more desirable than losing the flexibility for how versions of testrunner can mix with other tools.

In any case, I would envision that testrunner would run the same Maven plugin code before these testrunner changes when it is used with iDFlakies versions without your iDFlakies changes. For iDFlakies versions with your iDFlakies changes, testrunner would run the testrunner-core-plugin code. Having some duplicated code to enable testrunner to be more flexible is fine. As testrunner is a library used by other tools, we should refrain from deprecating functionality (e.g., APIs used by tools that just support Maven plugin) unless it is absolutely necessary and/or all tools (e.g., iFixFlakies) are no longer using such functionality

Reply:
Q: Can the testrunner changes not check if the plugin is a Maven plugin and then use the logger from AbstractMojo? Changing the format of the log output is quite undesirable
A: I will take a look for that.

Q: I would envision that testrunner would run the same Maven plugin code before these testrunner changes when it is used with iDFlakies versions without your iDFlakies changes.
A: Yes, I totally understand your idea. But I totally have no idea how to do it without changing iDFlakies code and I think it is impossible:

For example, some code in iDFlakies or iFixFlakies rely on org.apache.maven.project.MavenProject object to get build directory, version, classpaths and so on. But gradle plugin cannot provide MavenProject but can only provide org.gradle.api.Project object where the APIs are totally different from MavenProject. As iDFlakies and iFixFlakies plugins extend the TestPlugin class (https://github.com/TestingResearchIllinois/testrunner/blob/master/testrunner-maven-plugin/src/main/scala/edu/illinois/cs/testrunner/mavenplugin/TestPlugin.java ) where its execute() function only accepts MavenProject as the input, the interface itself cannot be compatible with gradle plugins. And as you can see, we modify its interface and make execute() accept our ProjectWrapper class.

Reply:
> For example, some code in iDFlakies or iFixFlakies rely on org.apache.maven.project.MavenProject object to get build directory, version, classpaths and so on. 
Can you elaborate on why this is a problem? I will repeat that the goal here is not for one iDFlakies version (e.g., a version with your iDFlakies PR changes) to work on older versions of testrunner, but the other way around, for one testrunner version (this version with your changes) to work with many versions of iDFlakies/iFixFlakies.

If a version of iDFlakies uses MavenProject, then treat it as a Maven plugin. These testrunner changes do NOT need to support Gradle projects on a version of iDFlakies that uses MavenProjects.

Reply:
Oh, I got your point. The backward compatibility code is in the newest commit (of courses, some duplicated code is added) and I have tested with the latest commit in idflakies without my changes. It works well.

Reply:
In addition, as for "Please do not change the logger that is used by the Maven plugin (e.g., the logger from AbstractMojo vs. java.util.logging.Logger)", since the backward compatibility code is added, shall we leave the original logger for old api but change the logger for new api?

Reply:
Please leave the original logger for old and new API. That is, regardless of iDFlakies version or testrunner version, the log should look similar and should both use the AbstractMojo logger.

Thanks a lot for the changes! The duplicated code should not a problem. I'll test your changes and review the code once again when the logger issue is addressed.

Reply:
Hi Wing, the logger modification is pushed into the last commit of both idflakies and testrunner. You can take a look now.